specs/Media.md

```diff
@@ -0,0 +1,146 @@
+            return true;
+        case IDM_IS_CURRENTLY_AUDIBLE:
+            BOOL isAudible;
+            m_webView->get_IsCurrentlyAudible(&isAudible);
+            MessageBox(nullptr, isAudible ? L"Yes" : L"No", L"Is Audible", MB_OK);
+            return true;
+    }
+}
+```
+
+## .NET and WinRT
+
+```c#
+    void MuteCmdExecuted(object target, ExecutedRoutedEventArgs e)
+    {
+        webView.CoreWebView2.Mute();
+    }
+
+    void UnmuteCmdExecuted(object target, ExecutedRoutedEventArgs e)
+    {
+        webView.CoreWebView2.Unmute();
+    }
+
+    void IsMutedCmdExecuted(object target, ExecutedRoutedEventArgs e)
+    {
+        bool is_muted = webView.CoreWebView2.IsMuted;
+        MessageBox.Show(this, is_muted? "Is muted": "Is Not muted", "Is Muted or Not");
+    }
+
+    void IsCurrentlyAudibleCmdExecuted(object target, ExecutedRoutedEventArgs e)
+    {
+        bool is_currently_audible = webView.CoreWebView2.IsCurrentlyAudible;
+        MessageBox.Show(this, is_currently_audible ? "Is audible" : "Is not audible", "Is Currently Audible or Not");
+    }
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2StagingIsMutedChangedEventHandler;
+interface ICoreWebView2StagingIsCurrentlyAudibleChangedEventHandler;
+
+[uuid(76eceacb-0462-4d94-ac83-423a6793775e), object, pointer_default(unique)]
+interface ICoreWebView2_2 : ICoreWebView2 {
+  /// Adds an event handler for the `IsMutedChanged` event.
+  /// `IsMutedChanged` runs when the mute state changes. The event may run 
+  /// when `Mute` and `Unmute` are called.
+  HRESULT add_IsMutedChanged(
+      [in] ICoreWebView2StagingIsMutedChangedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_IsMutedChanged`.
+  HRESULT remove_IsMutedChanged(
+      [in] EventRegistrationToken token);
+
+  /// Mutes all audio output from this CoreWebView2.
+  HRESULT Mute();
+  /// Unmutes all audio output from this CoreWebView2.
+  HRESULT Unmute();
+  /// Indicates whether all audio output from this CoreWebView2 is muted or not.
+  [propget] HRESULT IsMuted([out, retval] BOOL* isMuted);
+  
+  /// Adds an event handler for the `IsCurrentlyAudibleChanged` event.
+  /// `IsCurrentlyAudibleChanged` runs when the audible state changes.
+  HRESULT add_IsCurrentlyAudibleChanged(
+      [in] ICoreWebView2StagingIsCurrentlyAudibleChangedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_IsCurrentlyAudibleChanged`.
+  HRESULT remove_IsCurrentlyAudibleChanged(
+      [in] EventRegistrationToken token);
+  
+  /// Indicates whether any audio output from this CoreWebView2 is audible.
+  /// CoreWebView2 can be audible when muted, IsCurrentlyAudible is used to indicate
+  /// if there are audio/media currently playing.
```

Comment:
when you say 'audio/media' is that only media with audio? Like a video tag playing a video that has no audio (is that possible?)

Reply:
Looking at the comment below, it should only be any audio playing. I'm also trying to find a video with no audio to test that  

// Called when an audio change occurs to this WebContents. If |audible| is
  // true then one or more frames or child contents are emitting audio; if
  // false, then no frames or child contents are emitting audio. See
  // OnFrameAudioStateChanged for per-frame information.
virtual void OnAudioStateChanged(bool audible) {}

Reply:
I tried mute the video using the mute button within the video tag. If I mute the audio, the IsCurrentlyAudible value is false

Reply:
I added the IsCurrentlyAudible event hoping to help end developer detect whether content (media mainly) is playing, and they would be able to time out due to user inactive. But now thinking about it, IsCurrentlyAudible doesn't seems to be a good indication for that purpose. Also given the video tab behavior, the browser is treating IsCurrentlyAudible value as false when muted

Reply:
Best might be to remove this API for now and working on adding another API to indicate if content are playing so end developer can perform action due to user inactive. Would you think that would be better?

Reply:
IsCurrentlyAudible seems like its still useful for showing the end user which WebView2 is making noise and letting them then mute that WebView2 like the speaker icon in the tab name / favicon of browser tabs.